libcudacxx/include/cuda/std/__cccl/builtin.h

```diff
@@ -135,6 +147,32 @@
 #  define _CCCL_BUILTIN_EXPECT(...) __builtin_expect(__VA_ARGS__)
 #endif // _CCCL_CHECK_BUILTIN(builtin_expect)
 
+#if _CCCL_HAS_BUILTIN(__builtin_FILE) || defined(_CCCL_COMPILER_GCC) \
+  || (defined(_CCCL_COMPILER_MSVC) && _CCCL_MSVC_VERSION >= 1927)
+#  define _CCCL_BUILTIN_FILE() __builtin_FILE()
+#else // ^^^ _CCCL_HAS_BUILTIN(__builtin_FILE) ^^^ / vvv !_CCCL_HAS_BUILTIN(__builtin_FILE) vvv
+#  define _CCCL_BUILTIN_FILE() __FILE__
+#endif // !_CCCL_HAS_BUILTIN(__builtin_LINE)
+
+// NVCC below 11.3 cannot handle __builtin_FILE
+#if defined(_CCCL_CUDACC_BELOW_11_3)
+#  undef _CCCL_BUILTIN_FILE
+#  define _CCCL_BUILTIN_FILE() __FILE__
+#endif // _CCCL_CUDACC_BELOW_11_3
```

Comment:
Similarly with above, one condition should cover the two possibilities.

Reply:
Ditto that makes things easier